diff --git a/views_load_more.js b/views_load_more.js
index d13a665..6041c0d 100644
--- a/views_load_more.js
+++ b/views_load_more.js
@@ -55,7 +55,7 @@
 
     // Add the new content to the page.
     wrapper.find('.pager a').remove();
-    wrapper.find('.pager').parent('.item-list').html(new_content.find('.pager'));
+    wrapper.find('.pager').replaceWith(new_content.find('.pager'));
     wrapper.find(content_query)[method](new_content.find(content_query).children());
     if (effect.showEffect != 'show') {
       wrapper.find(content_query).children(':not(:visible)')[effect.showEffect](effect.showSpeed);
@@ -65,11 +65,21 @@
     wrapper.trigger('views_load_more.new_content', new_content.clone());
 
     // Attach all JavaScript behaviors to the new content
-    // Remove the Jquery once Class, TODO: There needs to be a better
-    // way of doing this, look at .removeOnce() :-/
-    var classes = wrapper.attr('class');
-    var onceClass = classes.match(/jquery-once-[0-9]*-[a-z]*/);
-    wrapper.removeClass(onceClass[0]);
+    // Remove the Jquery once classes
+    wrapper.removeClass(function() {
+      var i,
+        remove = '',
+        classes = this.className.split(' ');
+      for (i = 0; i < classes.length; i++) {
+        if (/-processed$/.test(classes[i])) {
+          if (remove != '') {
+            remove += ' ';
+          }
+          remove += classes[i];
+        }
+      }
+      return remove;
+    });
     var settings = response.settings || ajax.settings || Drupal.settings;
     Drupal.attachBehaviors(wrapper, settings);
   }
