diff --git a/simple_pass_reset.install b/simple_pass_reset.install
index 68a28be..af21cf6 100644
--- a/simple_pass_reset.install
+++ b/simple_pass_reset.install
@@ -13,3 +13,13 @@ function simple_pass_reset_install() {
    ->condition('name', 'simple_pass_reset', '=')
    ->execute();
 }
+
+/**
+ * Update our modules weight higer than system.module.
+ */
+function simple_pass_reset_update_7100() {
+ db_update('system')
+   ->fields(array('weight' => 1))
+   ->condition('name', 'simple_pass_reset', '=')
+   ->execute();
+}
\ No newline at end of file
